test/containerized_test.py

```diff
@@ -6,41 +6,126 @@
 
 import json
+import logging
+import os
+import stat
+import subprocess
+import time
+import shutil
+from collections import namedtuple
 import http.client as http_client
 
-from ovirt_imageio._internal import config
-from ovirt_imageio._internal import server
+import pytest
+
 from ovirt_imageio._internal.units import KiB
 
 from . import testutil
 from . import http
 
 
-TICKET_SIZE = 16 * KiB
+log = logging.getLogger("test")
 
+PODMAN_CMD = "podman"
+
+FILE_SIZE = 16 * KiB
+CONTAINER_IMG_PATH = "/image/disk.raw"
+CONTAINER_TICKET_PATH = "/ticket/file.json"
+CONTAINER_IMAGE = "localhost/ovirt-imageio:latest"
+
+Server = namedtuple("Server", ["host", "port"])
+Ticket = namedtuple("Ticket", ["id", "path"])
 
-def test_start_with_ticket(tmpdir):
-    # Create ticket
-    image = testutil.create_tempfile(tmpdir, name="disk.raw", size=TICKET_SIZE)
-    ticket = testutil.create_ticket(size=TICKET_SIZE, url=f'file://{image}')
-    ticket_id = ticket['uuid']
-    ticket_path = tmpdir.join("file.json")
-    ticket_path.write(json.dumps(ticket))
 
-    # Start server with ticket
-    cfg = config.load("test/conf/daemon.conf")
-    srv = server.Server(cfg, ticket=ticket_path)
-    srv.start()
+def _imageio_image_missing():
+    if shutil.which(PODMAN_CMD) is None:
+        return True
+    cmd = [PODMAN_CMD, "image", "inspect", CONTAINER_IMAGE]
     try:
-        data = b"a" * (TICKET_SIZE // 2) + b"b" * (TICKET_SIZE // 2)
-        # Test that we can upload
-        with http.RemoteClient(srv.config) as c:
-            res = c.put(f'/images/{ticket_id}', data)
-            assert res.status == http_client.OK
-        # Test that we can download and matches the uploaded data
-        with http.RemoteClient(srv.config) as c:
-            res = c.get(f'/images/{ticket_id}')
-            assert res.read() == data
-            assert res.status == http_client.OK
-    finally:
-        srv.stop()
+        return subprocess.check_call(
+            cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    except subprocess.CalledProcessError:
+        return True
+
+
+@pytest.fixture
+def tmp_image(tmpdir):
+    image_path = tmpdir.mkdir("image")
+    image = testutil.create_tempfile(
+        image_path, name=os.path.basename(CONTAINER_IMG_PATH), size=FILE_SIZE)
+    os.chmod(image, stat.S_IROTH | stat.S_IWOTH)
+    return str(image)
+
+
+@pytest.fixture
+def tmp_ticket(tmpdir):
+    ticket = testutil.create_ticket(
+        size=FILE_SIZE, url=f'file://{CONTAINER_TICKET_PATH}')
+    ticket_dir = tmpdir.mkdir("ticket")
+    ticket_path = ticket_dir.join(os.path.basename(CONTAINER_TICKET_PATH))
+    ticket_path.write(json.dumps(ticket))
+    os.chmod(ticket_path, stat.S_IROTH | stat.S_IWOTH)
+    return Ticket(ticket["uuid"], str(ticket_path))
+
+
+def _wait_for_server(host, port, timeout):
+    start = time.monotonic()
+    deadline = start + timeout
+    conn = http_client.HTTPConnection(host, port)
+    while True:
+        try:
+            conn.connect()
```

Comment:
This basically does tcp connect without sending any http request, so I wonder why our existing helper for waiting for a socket does not work.

Reply:
In my tests the existing helper gave false positives, passing the probing loop and then failing when actually sending the requests. Didn't look too much into it, as this solution seemed appropriate.